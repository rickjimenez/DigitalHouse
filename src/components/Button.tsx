import React, {FC} from 'react';
import {Text, TouchableOpacity, ViewStyle} from 'react-native';
import {useGlobalStyles} from '../styles/global';

interface Props {
  label: string;
  style?: ViewStyle;
  onPress?: () => void;
  testID?: string;
}

/**
 * Button Component
 *
 * @returns {React.ReactElement} Button.
 */
const Button: FC<Props> = ({
  label,
  style,
  onPress = () => {},
  testID,
}): JSX.Element => {
  const globalStyles = useGlobalStyles();
  return (
    <TouchableOpacity
      style={[globalStyles.btn, style]}
      onPress={onPress}
      testID={testID}>
      <Text style={globalStyles.btnTxt}>{label}</Text>
    </TouchableOpacity>
  );
};

export default Button;
