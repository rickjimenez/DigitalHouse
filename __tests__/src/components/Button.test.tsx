import 'react-native';
import {render, screen, fireEvent} from '@testing-library/react-native';
import React from 'react';
import Button from '../../../src/components/Button';

describe('test Button Comp', () => {
  test('renders correctly', () => {
    render(<Button label="test" />);

    const comp = screen.toJSON();
    expect(comp).toMatchSnapshot();
  });

  test('test press action', () => {
    const onPress = jest.fn();
    render(<Button label="test" onPress={onPress} testID="btn" />);
    const btn = screen.getByTestId('btn');
    fireEvent.press(btn);
    expect(onPress).toBeCalled();
  });
});
