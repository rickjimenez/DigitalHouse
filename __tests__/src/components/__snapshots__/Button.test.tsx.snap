// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test Button Comp renders correctly 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#334FFA",
      "borderRadius": 10,
      "height": 50,
      "justifyContent": "center",
      "opacity": 1,
    }
  }
>
  <Text
    style={
      {
        "color": "white",
        "fontSize": 14,
        "fontWeight": "800",
      }
    }
  >
    test
  </Text>
</View>
`;
